lib/ui.rb

```diff
@@ -79,7 +79,15 @@ def print(output)
         end
       end
 
+      def progress(output)
+        return if !$stdout.tty?
+
+        reset_line
+        print output
+      end
+
       def puts(output)
+        reset_line if $stdout.tty?
```

Comment:
This adds a reset of the line for every `puts` call, even if the line doesn't need to be cleaned. Would it be possible to instead clean the line at the end of the progress output?